### PR TITLE
add checks to calls to builtin make func

### PR DESCRIPTION
**Pull request by sjamesr**
Fixes https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2088.

### CODE REVIEW

**Comment by dlsniper** [comment]
Hi. Thanks for the PR. However, I think there's a problem with calling `getText()` from the `Annotator` implementations and I think this should be fixed in a different place. At least, if I recall this correctly, every time when calling `getText()` from it it triggers a PSI tree refresh, which wouldn't be a good idea. @zolotov / @ignatov please confirm I'm not remembering this wrong. If it's wrong, sorry for the noise.

**Comment by sjamesr** [comment]
If this is indeed a problem, I can move it to its own inspection.

**Comment by ignatov** [comment]
It's not a problem to get a text of local PSI element, but it's too expensive for stub element. In your case it looks good.

File: src/com/goide/highlighting/GoAnnotator.java [hunk 1]
```diff
@@ -80,13 +96,110 @@ else if (element instanceof GoTypeAssertionExpr) {
...
+
+    if (type instanceof GoSpecType) {
+      type = ((GoSpecType)type).getType();
+    }
+
+    return type;
+  }
+
+  private static boolean isIntegerType(GoType type) {
+    return intTypeNames.contains(type.getText());
```
Comment (ignatov) [inline src/com/goide/highlighting/GoAnnotator.java]: Type can be a stub, so it's definitely not a good idea to use getText.

Comment (sjamesr) [inline src/com/goide/highlighting/GoAnnotator.java]: Is there another way to find the type's name? Is this check simply not suitable for the GoAnnotator?

Comment (ignatov) [inline src/com/goide/highlighting/GoAnnotator.java]: Please use `GoPsiImplUtil#getTypeSpecSafe`

Comment (sjamesr) [inline src/com/goide/highlighting/GoAnnotator.java]: Can you give me an example of how I might use it here? For example, in the test I have

```go
type stringType string

make([]stringType, ...)
```

I determine the type of the first argument is a `GoArrayOrSliceType`, calling `getTypeSpecSafe` on that gives me `null`.

I'm unsure how to proceed.

**Comment by sjamesr** [comment]
@ignatov I think I've removed all calls to getText on stubbed elements. The error messages don't match Go's exactly, but I think it's fine.

**Comment by ignatov** [comment]
BTW, we have a special stub with text: com.goide.stubs.StubWithText

**Comment by ignatov** [comment]
I'm going to take a look to the PR today or tomorrow.

**Comment by sjamesr** [comment]
@ignatov regarding StubWithText, does this mean it's now safe to call getText on a GoType in GoAnnotator without triggering any lengthy re-parsing?

**Comment by ignatov** [comment]
For sure, calling getText on remote (resolved) element is potentially dangerous.

**Comment by sjamesr** [comment]
Ok, I was doing plenty of that previously. :) Hopefully this latest pull request is ready to go.

**Comment by ignatov** [comment]
Merged manually, thanks a lot for your help!

**Pull request closed without merging**